### PR TITLE
fix parameter_shorty match error

### PR DESCRIPTION
**Pull request by teble**
proto.shorty = result_shorty + param_shorty

### CODE REVIEW

File: dex_helper.cc [hunk 1]
```diff
@@ -646,7 +646,7 @@ bool DexHelper::IsMethodMatch(size_t dex_id, uint32_t method_id, uint32_t return
     const auto &proto = dex.ProtoIds()[method.proto_idx];
     const auto &shorty = strs[proto.shorty_idx];
     if (return_type != uint32_t(-2) && proto.return_type_idx != return_type) return false;
-    if (!parameter_shorty.empty() && shorty != parameter_shorty) return false;
+    if (!parameter_shorty.empty() && strcmp(parameter_shorty.data(), shorty.data() + 1) != 0) return false;
```
Comment (yujincheng08) [inline dex_helper.cc]: use c++ style

Comment (yujincheng08) [inline dex_helper.cc:649]: use c++ style

**Review by yujincheng08 — CHANGES_REQUESTED** [review]

File: dex_helper.cc [hunk 2]
```diff
@@ -646,7 +646,7 @@ bool DexHelper::IsMethodMatch(size_t dex_id, uint32_t method_id, uint32_t return
     const auto &proto = dex.ProtoIds()[method.proto_idx];
     const auto &shorty = strs[proto.shorty_idx];
     if (return_type != uint32_t(-2) && proto.return_type_idx != return_type) return false;
-    if (!parameter_shorty.empty() && strcmp(parameter_shorty.data(), shorty.data() + 1) != 0) return false;
+    if (!parameter_shorty.empty() && parameter_shorty.compare(shorty.data() + 1) != 0) return false;
```
Comment (yujincheng08) [inline dex_helper.cc:649]: still not c++ style

**Review by yujincheng08 — CHANGES_REQUESTED** [review]

File: dex_helper.cc (hunk 2 above)
Comment (teble) [reply dex_helper.cc:649]: what is c++ style, used substring?

Comment (yujincheng08) [reply dex_helper.cc:649]: `string_view::data()` does NOT necessarily end with `\0`, so try not to use it.

**Pull request closed without merging**